[command-reference]: Document what "auto" means in container run (#765)
apple/container: docs/command-reference.md

    *   `--virtualization`: Expose virtualization capabilities to the container (requires host and guest support)
*   **Registry options**
    *   `--scheme <scheme>`: Scheme to use when connecting to the container registry. One of (http, https, auto) (default: auto)
    
    * **Behavior of `auto`**  
  

>         When `auto` is selected, the target registry is considered **internal/local** based checking the host against several criteria:  

Reviewer: ```suggestion
        When `auto` is selected, the target registry is considered **internal/local** if the registry host matches any of these criteria:  
```
